.github/workflows/test-go.yml

```diff
@@ -412,4 +416,6 @@ jobs:
       - run: |
           ls -lhR test-results/go-test
           find test-results/go-test -mindepth 1 -mtime +3 -delete
+          // Prune invalid timing files
+          find . -maxdepth 1 -type f -name "test-results/go-test/*.json" -exec sh -c "jq . {} || rm {}" \; > /dev/null 2>&1
```

Comment:
If we have any invalid timing files for this run, do we still want to preserve the others, or might that have unforeseen results?

Reply:
Good question... I think it should be okay and will be useful data to keep around.

We could also move this prune up to the `test-go` job if you think it's risky, but I think @ryancragun is working on a fix to only persist on success.

As an aside, we're restoring the cache in this job, which means we'll have the older records as well and those should stick around as long as they're not malformed.